app/assets/javascripts/discourse/app/components/topic-map/topic-map-summary.gjs

```diff
@@ -1,162 +1,436 @@
 import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { hash } from "@ember/helper";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import { service } from "@ember/service";
 import { htmlSafe } from "@ember/template";
 import { gt } from "truth-helpers";
+import ConditionalLoadingSpinner from "discourse/components/conditional-loading-spinner";
 import DButton from "discourse/components/d-button";
-import RelativeDate from "discourse/components/relative-date";
+import TopicMapLink from "discourse/components/topic-map/topic-map-link";
 import TopicParticipants from "discourse/components/topic-map/topic-participants";
+import TopicViews from "discourse/components/topic-map/topic-views";
+import TopicViewsChart from "discourse/components/topic-map/topic-views-chart";
+import avatar from "discourse/helpers/bound-avatar-template";
 import number from "discourse/helpers/number";
-import slice from "discourse/helpers/slice";
+import { ajax } from "discourse/lib/ajax";
+import { emojiUnescape } from "discourse/lib/text";
+import dIcon from "discourse-common/helpers/d-icon";
 import i18n from "discourse-common/helpers/i18n";
-import { avatarImg } from "discourse-common/lib/avatar-utils";
+import I18n from "discourse-i18n";
+import DMenu from "float-kit/components/d-menu";
+
+const TRUNCATED_LINKS_LIMIT = 5;
+const MIN_POST_READ_TIME = 4;
+const MIN_LIKES_COUNT = 5;
+const MIN_PARTICIPANTS_COUNT = 5;
+const MIN_USERS_COUNT_FOR_AVATARS = 2;
+
+export const MIN_POSTS_COUNT = 10;
 
 export default class TopicMapSummary extends Component {
+  @service site;
+  @service siteSettings;
+  @service mapCache;
+  @service dialog;
+
+  @tracked allLinksShown = false;
+  @tracked top3LikedPosts = [];
+  @tracked views = [];
+  @tracked loading = true;
+
+  get shouldShowParticipants() {
+    return (
+      this.args.topic.posts_count >= MIN_POSTS_COUNT &&
+      this.args.topicDetails.participants?.length >=
+        MIN_USERS_COUNT_FOR_AVATARS &&
+      !this.site.mobileView
+    );
+  }
+
+  get first5Participants() {
+    return this.args.topicDetails.participants.slice(0, MIN_PARTICIPANTS_COUNT);
+  }
+
+  get readTimeMinutes() {
+    const calculatedTime = Math.ceil(
+      Math.max(
+        this.args.topic.word_count / this.siteSettings.read_time_word_count,
+        (this.args.topic.posts_count * MIN_POST_READ_TIME) / 60
+      )
+    );
+
+    return calculatedTime > 3 ? calculatedTime : null;
```

Comment:
Where is `3` coming from? If it's a magic number, it should probably be extracted to a `const`?